ci/test/build.sh

```diff
@@ -45,6 +45,36 @@ docker_run "cargo build --locked --release"
 ci_collapsed_heading "Building test binaries"
 docker_run "cargo test --locked --no-run && cargo test --locked --no-run --message-format=json > test-binaries.json"
 
+if true; then # XXX(brennan) replace this with the line below before landing, it is just here for testing my PR
+#if [[ "$BUILDKITE_BRANCH" = master ]]; then
+    ci_collapsed_heading "Building .deb package"
+    # shellcheck disable=SC2016
+    # TODO (brennan) - fix this version number
+    docker_run 'cargo-deb --no-strip --deb-version 0.1.0-$(git rev-list HEAD | wc -l)-$(git rev-parse HEAD) -p materialized -o target/debian/materialized.deb'
```

Comment:
You extract a `COMMIT_INDEX` and `COMMIT_HASH` below; probably should use those here, too.

By the way, convention for local variables is lowercase: `commit_index` and `commit_hash`. Only environment variables get SCREAMING CASE.